### ISSUE TITLE
Unload Model : how?

### ISSUE THREAD

**Issue by Khampol** [stavsap/comfyui-ollama#62]
Hello, 
I need sometime to unload model like this [command](https://github.com/ollama/ollama/blob/main/docs/faq.md#how-do-i-keep-a-model-loaded-in-memory-or-make-it-unload-immediately), how I could do that ?

**Comment by bigcat88** [comment]
send an empty prompt with keepalive zero.

**Comment by MoreColors123** [comment]
I also had this problem, and i don't remember if bigcat88's solved it. Had to look around for a while and then found/made a simple script with ChatGPT I think:

Put this in a file called "unload ollama.ps1":

# Define the IP addresses
$ips = @("127.0.0.1")

# Function to fetch models from a given IP address
function Fetch-Models {
    param (
        [string]$ip
    )
    $json_response = Invoke-RestMethod -Uri "http://${ip}:11434/api/ps"
    $models = $json_response.models | ForEach-Object { 
        "$ip $($_.model)" 
    }
    return $models
}

# Step 1: Fetch models from all IP addresses and display as a numbered list
$models_list = @()
$index = 1
Write-Host "Available models on Server ${ip}:"
foreach ($ip in $ips) {
    $models = Fetch-Models -ip $ip
    foreach ($model in $models) {
        $models_list += $model
        Write-Host "$index) $($model -split ' ')[1]"
        $index++
    }
}

# Step 2: Let the user select a model by number
$model_num = Read-Host "Enter the model number to unload a Model or 0 to exit"
if ($model_num -eq 0) {
    exit
}

# Step 3: Get the selected IP and model
$selected_entry = $models_list[$model_num - 1].Trim()
$selected_ip = ($selected_entry -split ' ')[0].Trim()
$selected_model = ($selected_entry -split ' ')[1].Trim()

# Step 4: Run the final Invoke-RestMethod command with the selected IP and model
$body = @{ model = $selected_model; keep_alive = 0 } | ConvertTo-Json
Invoke-RestMethod -Uri "http://${selected_ip}:11434/api/generate" -Method Post -Body $body -ContentType "application/json"
Write-Host "$selected_model on IP $selected_ip unloaded"



----------------

Now Put this in a .bat file in the same folder:

@echo off
PowerShell -NoProfile -ExecutionPolicy Bypass -File "unload ollama.ps1"
pause

---------

Now set a Windows shortcut to this .bat file via the properties menu (or double click the .bat of course)

**Comment by Khampol** [comment]
> send an empty prompt with keepalive zero.

noted.

**Comment by Khampol** [comment]
> I also had this problem, and i don't remember if bigcat88's solved it. Had to look around for a while and then found/made a simple script with ChatGPT I think:
> 
> Put this in a file called "unload ollama.ps1":
> 
> ```
> # Define the IP addresses
> $ips = @("127.0.0.1")
> 
> # Function to fetch models from a given IP address
> function Fetch-Models {
>     param (
>         [string]$ip
>     )
>     $json_response = Invoke-RestMethod -Uri "http://${ip}:11434/api/ps"
>     $models = $json_response.models | ForEach-Object { 
>         "$ip $($_.model)" 
>     }
>     return $models
> }
> 
> # Step 1: Fetch models from all IP addresses and display as a numbered list
> $models_list = @()
> $index = 1
> Write-Host "Available models on Server ${ip}:"
> foreach ($ip in $ips) {
>     $models = Fetch-Models -ip $ip
>     foreach ($model in $models) {
>         $models_list += $model
>         Write-Host "$index) $($model -split ' ')[1]"
>         $index++
>     }
> }
> 
> # Step 2: Let the user select a model by number
> $model_num = Read-Host "Enter the model number to unload a Model or 0 to exit"
> if ($model_num -eq 0) {
>     exit
> }
> 
> # Step 3: Get the selected IP and model
> $selected_entry = $models_list[$model_num - 1].Trim()
> $selected_ip = ($selected_entry -split ' ')[0].Trim()
> $selected_model = ($selected_entry -split ' ')[1].Trim()
> 
> # Step 4: Run the final Invoke-RestMethod command with the selected IP and model
> $body = @{ model = $selected_model; keep_alive = 0 } | ConvertTo-Json
> Invoke-RestMethod -Uri "http://${selected_ip}:11434/api/generate" -Method Post -Body $body -ContentType "application/json"
> Write-Host "$selected_model on IP $selected_ip unloaded"
> ```
> 
> Now put this in a .bat file in the same folder:
> 
> ```
> @echo off
> PowerShell -NoProfile -ExecutionPolicy Bypass -File "unload ollama.ps1"
> pause
> ```
> 
> Now set a Windows shortcut to this .bat file via the properties menu (or double click the .bat of course)

A bat file?... And how you suppose to use it ? Manually ?

**Comment by MoreColors123** [comment]
Yes exactly. If you want to make a shortcut (in win) you actually need to make a link to the bat file and then set a keyboard shortcut by right-clicking on the link -> properties. At least this works for me.

**Comment by Khampol** [comment]
ok noted that. thanks.

**Comment by Khampol** [comment]
Guys, I want just to add to this topic that I use an appli (DIY) that can upload (close and reopen ollama svr) in on clic. 
Voila :>
